sale_report_delivery_date/reports/sale_report.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2022 Camptocamp SA (https://www.camptocamp.com).
+# @author Iván Todorovich <ivan.todorovich@camptocamp.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class SaleReport(models.Model):
+    _inherit = "sale.report"
+
+    delivery_date = fields.Datetime(readonly=True)
+
+    def _query(self, with_clause="", fields=None, groupby="", from_clause=""):
+        if not fields:
+            fields = {}
```

Comment:
You don't use fields, then I think it's useless to set `fields` with empty dict:
```suggestion
```

Reply:
[There's a bug](https://github.com/odoo/odoo/blob/a7df631c9bab75767c8724e42aa2f742d032dce9/addons/sale_stock/report/sale_report.py#L12-L13) in core `sale_stock` module :(
It crashed if I don't add this one, but I'll remove the other

```python-traceback
Traceback (most recent call last):
  File "/opt/odoo/odoo/service/server.py", line 1201, in preload_registries
    registry = Registry.new(dbname, update_module=update_module)
  File "/opt/odoo/odoo/modules/registry.py", line 89, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/opt/odoo/odoo/modules/loading.py", line 461, in load_modules
    loaded_modules, update_module, models_to_check)
  File "/opt/odoo/odoo/modules/loading.py", line 349, in load_marked_modules
    perform_checks=perform_checks, models_to_check=models_to_check
  File "/opt/odoo/odoo/modules/loading.py", line 199, in load_module_graph
    registry.init_models(cr, model_names, {'module': package.name}, new_install)
  File "/opt/odoo/odoo/modules/registry.py", line 406, in init_models
    model.init()
  File "/opt/odoo/addons/sale/report/sale_report.py", line 157, in init
    self.env.cr.execute("""CREATE or REPLACE VIEW %s as (%s)""" % (self._table, self._query()))
  File "/mnt/data/odoo-addons-dir/sale_report_delivery_date/reports/sale_report.py", line 34, in _query
    from_clause=from_clause,
  File "/opt/odoo/addons/sale_stock/report/sale_report.py", line 13, in _query
    fields['warehouse_id'] = ", s.warehouse_id as warehouse_id"
TypeError: 'NoneType' object does not support item assignment
```